app/code/Magento/Catalog/Model/ResourceModel/Category/Collection.php

```diff
@@ -329,7 +336,7 @@ public function loadProductCount($items, $countRegular = true, $countAnchor = tr
             foreach ($anchor as $item) {
                 $productsCount = isset($categoryProductsCount[$item->getId()])
                     ? (int)$categoryProductsCount[$item->getId()]
-                    : $this->getProductsCountFromCategoryTable($item, $websiteId);
+                    : 0;
```

Comment:
I would like to have a comment here noting the `0` will be used when the index hasn't been built yet.  
The PR notes explained it nicely, but I can imagine someone (me?) stumbling over this in future and wondering about the reasoning behind it.